src/Orleans.Transactions/InClusterTM/ActiveTransactionsTracker.cs

```diff
@@ -0,0 +1,136 @@
+        }
+
+        public void Start(long initialTransactionId)
+        {
+            smallestActiveTransactionId = initialTransactionId + 1;
+            highestActiveTransactionId = initialTransactionId;
+            maxAllocatedTransactionId = initialTransactionId;
+
+            allocationThread.Start();
+        }
+
+        public long GetNewTransactionId()
+        {
+            var id = Interlocked.Increment(ref highestActiveTransactionId);
+
+            if (maxAllocatedTransactionId - highestActiveTransactionId <= config.AvailableTransactionIdThreshold)
+            {
+                // Signal the allocation thread to allocate more Ids
+                allocationEvent.Set();
+            }
+
+            while (id > maxAllocatedTransactionId)
+            {
+                // Wait until the allocation thread catches up before returning.
+                // This should never happen if we are pre-allocating fast enough.
+                allocationEvent.Set();
+                lock (lockObj)
+                {
+                }
+            }
+
+            return id;
+        }
+
+        public long GetSmallestActiveTransactionId()
+        {
+            // NOTE: this result is not strictly correct if there are NO active transactions
+            // but for all purposes in which this is used it is still valid.
+            // TODO: consider renaming this or handling the no active transactions case.
+            return Interlocked.Read(ref smallestActiveTransactionId);
+        }
+
+        public long GetHighestActiveTransactionId()
+        {
+            // NOTE: this result is not strictly correct if there are NO active transactions
+            // but for all purposes in which this is used it is still valid.
+            // TODO: consider renaming this or handling the no active transactions case.
+            lock (lockObj)
+            {
+                return Math.Min(highestActiveTransactionId, maxAllocatedTransactionId);
+            }
+        }
+
+
+        public void PopSmallestActiveTransactionId()
+        {
+            Interlocked.Increment(ref smallestActiveTransactionId);
+        }
+
+        private void AllocateTransactionId(object args)
+        {
+            while (!this.disposed)
+            {
+                try
+                {
+                    allocationEvent.WaitOne();
+                    if (this.disposed) return;
+
+                    lock (lockObj)
+                    {
+                        if (maxAllocatedTransactionId - highestActiveTransactionId <= config.AvailableTransactionIdThreshold)
+                        {
+                            var batchSize = config.TransactionIdAllocationBatchSize;
+                            transactionLog.UpdateStartRecord(maxAllocatedTransactionId + batchSize).GetAwaiter().GetResult();
+
+                            maxAllocatedTransactionId += batchSize;
+                        }
+                    }
+                }
+#if !NETSTANDARD
```

Comment:
not needed any more, this is now supported in .NET Standard 2.0 (and the NETSTANDARD compilation constant is no longer defined in our solution anyway, only BUILD_FLAVOR_LEGACY could be used to differentiate between the 2 build flavors)